pkg/resource/deploy/step_executor.go

```diff
@@ -407,7 +407,12 @@ func (se *stepExecutor) cancelDueToError(err error, step Step) {
 func (se *stepExecutor) executeStep(workerID int, step Step) error {
 	var payload interface{}
 	events := se.deployment.events
-	if events != nil {
+
+	// DiffSteps are special, we just use them for step worker parallelism but they shouldn't be passed to the rest of
+	// the system.
```

Comment:
I guess when refresh lands we'll generalise this to something like `isYieldingStep` or the like -- is that the idea?